01-indoor-floor-level-tracker/web-app/src/services/AppModel.ts

```diff
@@ -4,6 +4,14 @@ export type ProjectID = DomainModel.ProjectID;
 export type FleetID = DomainModel.FleetID;
 export type DeviceID = DomainModel.DeviceID;
 
+export interface AuthToken {
+  access_token?: string;
+  scope?: string;
+  expires_in?: number;
+  token_type?: string;
+  token_expiration_date?: string;
```

Comment:
Remind me the difference between AppModel and DomainModel?

Do you need to store `expires_in` and `token_expiration_date`? Seems like expires_in will start to go stale as soon as you store it whereas token_expiration_date will be accurate forever.

Reply:
AppModel is client side, app-specific objects, domain model is server side, more generic objects that you could fetch from a database, Notehub API, etc.

This object goes from server side to client side and back again, so I added it to `AppModel` but the case could be made for either file.

`expires_in` is one of the properties the token comes generated at creation, `token_expiration_date` is a property I manually calculate based on the current time and `expires_in`. And you're correct, it goes stale after initial generation.

I'm planning to ask the Notehub team if they could include something like a timestamp akin to `token_expiration_date` in addition to `expires_in` so we don't have to do the math for it.